doc/Customization-guide/Code-style/C-C++/index.md

```diff
@@ -0,0 +1,191 @@
+- Use 100-character column width.
+- Use `snake_case` for the names (not `camelCase`) in C code.
+- All macro names must be in UPPERCASE.
+- Use [1TBS](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS) (the one true brace style).
+
+  ```cpp
+  int main(int argc, char *argv[])
+  {
+     if (argc > 2) {
+         printf("too many args\n");
+     } else {
+         printf("too little args\n");
+     }
+
+     return 0;
+  }
+  ```
+
+  Note brace position.
+
+- The braces in control structures are required even for a single statement.
+- The pointer star must be aligned to the variable name.
+
+  ```cpp
+  int *x;
+  ```
+
+- Labels must be indented one level less than the normal indentation (except for case labels).
+
+  ```cpp
+  int main(int argc, char *argv[])
+  {
+      switch (argc) {
+          case 3:
+              printf("hi\n");
+
+          default:
+              goto fail;
+      }
+
+      return 0;
+
+  fail:
+      return 1;
+  }
+  ```
+
+- Prefer double-indent instead of alignment in conditions and argument lists.
+
+  ```cpp
+  int my_function(int param1, int param2,
+         int param3, int param4)
+  {
+      if (param1 > 3 && param2 > 3 &&
+              param3 > 3 && param4 > 3) {
+          return param1 + param2 + param3 +
+                  param4;
+      } else {
+          return 0;
+      }
+  }
+  ```
+
+- Don't allow trailing spaces or lines.
+- Wrap complex macro in a `do { } while(0)` statement
+
+  ```c
+  #define COMPLEX_MACRO(a, b, c) \
+      do { foo(a); bar(b); baz(c); } while(0)
+  ```
+
+- Place comma on the same line as parameter name when wrap lines due to long function or parameter names
+
+  ```c
+  // Good
+  void function_with_many_params_and_long_name(long_name_param a,
+          long_name_param b, long_name_param c, long_name_param d,
+          long_name_param e, long_name_param f)
+
+
+  // Bad
+  void function_with_many_params_and_long_name(param a
+                                              , param b
+                                              , param c)
+  ```
+
+- Commented out code is not allowed without great reason and notes.
+Use git instead.
+
+- Do not use `_t` suffix for type names.
+It is reserved by the compiler.
+
+- Avoid using TODO or FIXME.
+If still necessarry, create an issue and place TODO in the code along with issue number and description as shown below.
+
+  ```c
+  // TODO(KAA-xxx): short description
+  ```
+
+# Semantics
+
+- Initialize all variables.
+- Prefer declaration with initialization.
+
+  ```c
+  int a = 10; // Good
+  int b; // Bad
+  ```
+
+- Initiailize structures with members declaration
+
+  ```c
+  struct st {
+      int a;
+      int b;
+      int c;
+  };
+
+
+  void foo(void)
+  {
+      struct st obj = {
+          .a = 1,
+          .b = 2,
+          .c = 3,
+      };
+  }
+
+  ```
+
+- Typedefs that indicate size and signedness should be used in place of the basic numerical types.
+E.g. `size_t` for holding objects sizes.
```

Comment:
`size_t` is a different beast from `Typedefs that indicate size and signedness`. The later are `uint32_t`, int8_t`, etc.